### PR TITLE
Tags in URLs

### PR DESCRIPTION
**Pull request by quadrismegistus**
For https://github.com/Princeton-CDH/lenape-timetree/issues/114

### CODE REVIEW

**Comment by quadrismegistus** [comment]
Here's a working version. Let me know if the `#?` part is too ugly. But the `#` ensures we don't leave page.

File: assets/js/leaves.js [hunk 1]
```diff
@@ -138,6 +144,15 @@ class Leaf {
     }
   }
 
+  static selectLeavesByTag() {
+    // If the page is loaded with a tag link, select those leaves
+    if (location.search.startsWith("?tag=")) {
```
Comment (rlskoeser) [inline assets/js/leaves.js:149]: This is order dependent and we might add other params; let's parse as a `URL` first so we can check with `params` object

File: assets/js/leaves.js [hunk 2]
```diff
@@ -64,6 +64,12 @@ class Leaf {
     for (let item of leaves) {
       item.classList.add(Leaf.selectedClass);
     }
+
+    // encode URL to reflect tag
+    let slug = encodeURIComponent(tag);
+    let url = new URL(document.location.href.replace("#", ""));
```
Comment (rlskoeser) [inline assets/js/leaves.js:70]: why does the hash need to be replaced here? (might warrant a comment)

**Review by rlskoeser — COMMENTED** [review]
Your updates look good! This is working better.

I just pushed some changes to the template and leaf json data so we can use tag slugs — I made more changes than I intended because I wanted to be sure it was working properly.

Please review my changes and see if it makes sense and if you want to make any adjustments. 

We could add logic to deselect active leaf when we close the panel — although technically I guess that's part of #110, so it could wait. 

How do you feel about taking a stab at a unit test for your new `selectLeavesByTag` method?

**Comment by rlskoeser** [comment]
> We could add logic to deselect active leaf when we close the panel — although technically I guess that's part of #110, so it could wait.

Clarification on this: right now we deselect all leaves but we don't clear the tag param from the url, which is confusing. I don't feel strongly about needing to fix it here, though.

**Comment by quadrismegistus** [comment]
> > We could add logic to deselect active leaf when we close the panel — although technically I guess that's part of #110, so it could wait.
> 
> Clarification on this: right now we deselect all leaves but we don't clear the tag param from the url, which is confusing. I don't feel strongly about needing to fix it here, though.

This should be taken care of here now.

**Comment by rlskoeser** [comment]
documenting dev steps for the revised approach we agreed on in our whiteboarding session last week:

- `deselectAll` method should be removed 
- new `updateSelection` method that sets highlighting in the tree based on the url (one/both/neither of tag param & leaf hash); this method should be called when the page is loaded 
- select by tag / leaf - rename to `setCurrentLeaf` / `setCurrentTag` (or similar) ; logic is to update url (param or hash) and then call `updateSelection`
- `closeLeafDetails` - unset current leaf and tag, then call `updateSelection`; I think the proposed approach we discussed was to unset by calling `setCurrentLeaf` / `setCurrentTag` with a null value or similar
- `closeTag` - unset current tag and then call `updateSelection` 
- add unit tests for the new methods including interactions between these parameters (both, none, either one)

Additionally: let's create a new branch `feature/tag-interaction` (or similar) and then merge this and the other tag-related PR into that branch. Dev review can happen before merging into that branch, acceptance testing can happen from a new PR to merge all the tag interactions into `develop`.

**Comment by quadrismegistus** [comment]
Yeah I think we will want to abandon this branch in favor of https://github.com/Princeton-CDH/lenape-timetree/pull/141

**Comment by rlskoeser** [comment]
superseded by #141
